Crash on MacOS

Hi !

I've been trying to run the latest version on Mac 10.15.7 but it crashes immediately.
Only info I managed to get is this:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x477f9aa]

goroutine 46 [running]:
wombat/internal/app.(*api).GetWorkspaceOptions(0xc0004d8000, 0x4a1a4be, 0xb, 0xc0003701e0)
	/Users/runner/work/wombat/wombat/internal/app/api.go:174 +0x14a
wombat/internal/app.(*api).wailsReady(0xc0004d8000, 0x0, 0x0, 0x0)
	/Users/runner/work/wombat/wombat/internal/app/api.go:96 +0x157
created by github.com/wailsapp/wails/lib/event.(*Manager).Start.func1
	/Users/runner/go/pkg/mod/github.com/wailsapp/wails@v1.9.1/lib/event/manager.go:124 +0x457
```

Previous version runs flawlessly.
Let me know if there's something else I can do to give you more debug information.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,9 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+### Fixed
+- Crash on launch due to poor handling of the default workspace for first time user
+
 ## [v0.3.1] - 2020-11-20
 
 ### Added

--- a/frontend/src/views/WorkspaceList.svelte
+++ b/frontend/src/views/WorkspaceList.svelte
@@ -53,7 +53,7 @@
             <span class="box"/>
           {/if}
         </td>
-        <td class:selected={wksp.id === current.id}>{wksp.addr}</td>
+        <td class:selected={wksp.id === current.id}>{!wksp.addr && wksp.id === "wksp_default" ? "default workspace" : wksp.addr}</td>
         <td>
           {#if wksp.id !== "wksp_default"}
             <CrossButton on:click={() => onDeleteClicked(wksp)} color={isWin ? "#bf616a" : "var(--red-color)"} />

--- a/internal/app/api.go
+++ b/internal/app/api.go
@@ -162,20 +162,30 @@ func (a *api) getCurrentState() *workspaceState {
 
 // GetWorkspaceOptions gets the workspace options from the store
 func (a *api) GetWorkspaceOptions() (*options, error) {
-	val, err := a.store.get([]byte(a.state.CurrentID))
+	wo := &options{
+		ID: a.state.CurrentID,
+	}
+
+	val, err := a.store.get([]byte(wo.ID))
+	if err != nil && err != errKeyNotFound {
+		return nil, err
+	}
+
+	if len(val) == 0 {
+		return wo, nil
+	}
+
+	dec := gob.NewDecoder(bytes.NewBuffer(val))
+	err = dec.Decode(wo)
 	if err != nil {
 		return nil, err
 	}
 
-	var opts *options
-	dec := gob.NewDecoder(bytes.NewBuffer(val))
-	err = dec.Decode(&opts)
-
-	if opts.ID == "" {
-		opts.ID = defaultWorkspaceKey
+	if wo.ID == "" {
+		wo.ID = defaultWorkspaceKey
 	}
 
-	return opts, err
+	return wo, nil
 }
 
 // GetReflectMetadata gets the reflection metadata from the store by addr
@@ -211,6 +221,7 @@ func (a *api) ListWorkspaces() ([]options, error) {
 		return nil, err
 	}
 	var opts []options
+	hasDefault := false
 	for _, val := range items {
 		opt := options{}
 		dec := gob.NewDecoder(bytes.NewBuffer(val))
@@ -219,11 +230,15 @@ func (a *api) ListWorkspaces() ([]options, error) {
 		}
 		// opts.ID was added in v0.3.0 so need to double check
 		if opt.ID == defaultWorkspaceKey || opt.ID == "" {
+			hasDefault = true
 			opt.ID = defaultWorkspaceKey
 			opts = append([]options{opt}, opts...)
 			continue
 		}
 		opts = append(opts, opt)
+	}
+	if !hasDefault {
+		opts = append([]options{options{ID: defaultWorkspaceKey}}, opts...)
 	}
 	return opts, nil
 }

